script.py

```diff
@@ -172,13 +176,15 @@ def _add_track_to_sanitized_list(tracks):
             sanitized_tracks.append([
                 track['name'], track['artists'][0]['name']
             ])
-
-    tracks_catch = tracks if tracks else sp.user_playlist(spotify_id, playlist['id'], fields="tracks,next")['tracks']
-    _add_track_to_sanitized_list(tracks)
-    while tracks['next']:
-        tracks = sp.next(tracks)
-        _add_track_to_sanitized_list(tracks)
-
+    
+    if (playlist['owner']['id'] == spotify_id):
```

Comment:
Do you mind removing this if statement and just having this line:

```
tracks_catch = tracks if tracks else sp.user_playlist(playlist['owner']['id'], playlist['id'], fields="tracks,next")['tracks']
```

Since we don't need to check if the playlist creator ID is the same as the user ID.